### PR TITLE
youtube: add support for OAuth2 tokens

### PR DESCRIPTION
**Pull request by dumbmoron**

### CODE REVIEW

File: src/modules/sub/generate-youtube-tokens.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    )
+);
+
+tube.session.once('auth-error', (err) => bail('An error occurred:', err));
+tube.session.once('auth', ({ status, credentials, ...rest }) => {
+    if (status !== 'SUCCESS') {
+        bail('something went wrong', rest);
+    }
+
+    console.log(credentials);
```
Comment (wukko) [inline src/modules/sub/generate-youtube-tokens.js:23]: is this needed to be logged? if not, please remove it

File: src/modules/sub/generate-youtube-tokens.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    console.log(
```
Comment (wukko) [inline src/modules/sub/generate-youtube-tokens.js:25]: i'd add a message like "add this cookie to youtube_oauth array in your cookies file:"

**Comment by wukko** [comment]
i think generation script should be moved to a different directory, something like `src/tooling`

**Comment by wukko** [comment]
i'd also show a message that access token is sensitive info and shouldn't be shown to or shared with anyone

File: package.json [hunk 3]
```diff
@@ -11,9 +11,9 @@
     "scripts": {
         "start": "node src/cobalt",
         "setup": "node src/modules/setup",
-        "test": "node src/test/test",
+        "test": "node src/util/test",
         "build": "node src/modules/buildStatic",
-        "testFilenames": "node src/test/testFilenamePresets"
+        "testFilenames": "node src/util/testFilenamePresets"
     },
```
Comment (wukko) [inline package.json:17]: why not add a script shortcut for generating a youtube token? :3c

Comment (dumbmoron) [reply package.json:17]: idk i dont think its really necessary here

Comment (dumbmoron) [reply package.json:17]: testFilenames should probably get removed too

Comment (wukko) [reply package.json:17]: it's either both or neither i think

Comment (wukko) [reply package.json:17]: i'd remove test filenames but add youtube token generate script tbh

**Review by wukko — APPROVED** [review]
lgtm

**Comment by wukko** [comment]
does not fully close #551, but improves the situation by a lot

**Pull request merged**